src/bin/ion/commands/inspect.rs

```diff
@@ -187,22 +187,39 @@ impl<'a, 'b> IonInspector<'a, 'b> {
         Ok(inspector)
     }
 
+    fn confirm_encoding_is_supported(&self, encoding: IonEncoding) -> Result<()> {
+        use IonEncoding::*;
+        match encoding {
+            Text_1_0 | Text_1_1 => {
+                bail!("`inspect` does not support text Ion streams.");
+            }
+            Binary_1_0 => Ok(()),
+            Binary_1_1 => bail!("`inspect` does not yet support binary Ion v1.1"),
+            // `IonEncoding` is #[non_exhaustive]
+            _ => bail!("`inspect does not yet support {}", encoding.name()),
+        }
+    }
+
     /// Iterates over the items in `reader`, printing a table section for each top level value.
     fn inspect_top_level<Input: IonInput>(
         &mut self,
         reader: &mut SystemReader<AnyEncoding, Input>,
     ) -> Result<()> {
         const TOP_LEVEL_DEPTH: usize = 0;
-        self.write_table_header()?;
```

Comment:
🗺️ We now defer writing the table header until after we've read the first item from the stream and confirmed it's of a supported encoding.